### PR TITLE
feat(conf/poller): Added APIs to generate and reload monitoring server configuration

### PR DESCRIPTION
**Pull request by callapa**
## Description

Added 6 new APIs to generate and reload monitoring server configuration.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

See the API documentation in the **Monitoring Server** section

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by adr-mo — DISMISSED** [review]

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: composer.json [hunk 1]
```diff
@@ -64,10 +63,14 @@
...
-        "symfony/serializer-pack": "^1.0",
         "symfony/translation": "4.4.*",
         "symfony/validator": "4.4.*",
-        "symfony/yaml": "4.4.*"
+        "symfony/yaml": "4.4.*",
+        "doctrine/annotations": "^1.0",
+        "phpdocumentor/reflection-docblock": "^5.2",
+        "symfony/property-access": "4.4.*",
+        "symfony/property-info": "4.4.*",
+        "symfony/serializer": "4.4.*"
```
Comment (kduret) [inline composer.json:73]: could you use alphabetical order ? It will be easier to read

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16238,4 +16238,40 @@ msgid "Contact Details"
...
+msgstr "La requête a échoué"
+
+msgid "Response empty"
+msgstr "Réponse vide"
+
+msgid "Unable to load the Symfony container"
+msgstr "Impossible de charger le conteneur Symfony"
+
+msgid "Error on monitoring server #%d: %s"
+msgstr "Erreur sur le serveur de supervision #%d : %s"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16274]:
```suggestion
msgstr "Erreur sur le serveur de supervision #%d: %s"
```

File: src/Centreon/Application/Controller/Configuration/MonitoringServerController.php [hunk 3]
```diff
@@ -52,26 +61,167 @@ public function __construct(MonitoringServerServiceInterface $monitoringServerSe
...
      * @param RequestParametersInterface $requestParameters
-     * @param Request $request
      * @return View
      * @throws \Exception
      */
-    public function findServer(RequestParametersInterface $requestParameters, Request $request): View
+    public function findServer(RequestParametersInterface $requestParameters): View
     {
         $this->denyAccessUnlessGrantedForApiConfiguration();
         $server = $this->monitoringServerService->findServers();
```
Comment (kduret) [inline src/Centreon/Application/Controller/Configuration/MonitoringServerController.php:73]: `findServers` should be renamed to `findServer` ?

File: src/Centreon/Domain/MonitoringServer/UseCase/GenerateConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * @var MonitoringServerRepositoryInterface
+     */
+    private $monitoringServerRepository;
+
+    /**
+     * @var MonitoringServerConfigurationRepositoryInterface
+     */
+    private $repository;
```
Comment (kduret) [inline src/Centreon/Domain/MonitoringServer/UseCase/GenerateConfiguration.php:49]: may need to be renamed to more explicit name

File: src/Centreon/Domain/MonitoringServer/UseCase/ReloadAllConfigurations.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ * @package Centreon\Domain\MonitoringServer\UseCase
+ */
+class ReloadAllConfigurations
+{
+    use LoggerTrait;
+
+    /**
+     * @var MonitoringServerConfigurationRepositoryInterface
+     */
+    private $repository;
```
Comment (kduret) [inline src/Centreon/Domain/MonitoringServer/UseCase/ReloadAllConfigurations.php:43]: may need to be renamed to more explicit name

File: src/Centreon/Domain/MonitoringServer/UseCase/ReloadConfiguration.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * @var MonitoringServerRepositoryInterface
+     */
+    private $monitoringServerRepository;
+
+    /**
+     * @var MonitoringServerConfigurationRepositoryInterface
+     */
+    private $repository;
```
Comment (kduret) [inline src/Centreon/Domain/MonitoringServer/UseCase/ReloadConfiguration.php:48]: may need to be renamed to more explicit name

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16238,4 +16238,40 @@ msgid "Contact Details"
...
+msgstr "Erreur lors de la récupération des serveurs de supervision"
+
+msgid "Authentication token expired"
+msgstr "Le jeton d'authentification a expiré"
+
+msgid "Authentication token not found"
+msgstr "Jeton d'authentification non trouvé"
+
+msgid "Request failed"
+msgstr "La requête a échoué"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16265]:
```suggestion
msgid "Request failed (%d)"
msgstr "La requête a échoué (%d)"
```

**Review by kduret — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**